webrender/src/clip_scroll_node.rs

```diff
@@ -412,6 +411,38 @@ impl ClipScrollNode {
             _ => false,
         }
     }
+
+    pub fn find_unclipped_rectangle(&self,
```

Comment:
It looks like `MaskBounds::inner` is using `inner_rect_safe()` while your optimization is using `inner_rect_full()`. I'm not sure what is correct here, so would it make sense to try to combine these calculations in a later patch?